pkg/test/nodes.go

```diff
@@ -68,14 +69,17 @@ func Node(overrides ...NodeOptions) *v1.Node {
 }
 
 func NodeClaimLinkedNode(nodeClaim *v1beta1.NodeClaim) *v1.Node {
+	taints := lo.Filter(nodeClaim.Spec.StartupTaints, func(t v1.Taint, _ int) bool {
```

Comment:
This filters out the taint from the NodeClaim startup taints, but I'm curious if we should have this taint be added to nodes always to properly mock the true behavior of what happens from a CloudProvider and then orchestrate the removal from the controller.

This also prevents you having to add it in the StartupTaints section of the NodeClaim

Reply:
Removed.